website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,464 @@
+
+If we try to run our program now...
+
+![A really messed up quad!](../../../images/opengl/chapter1/lesson3/messed-up-quad.png)
+
+Well, that doesn't look like the quad we were expecting. This is because we have updated the vertex buffer we're passing to the vertex shader,
+but haven't updated anything else such as our vertex layout definition. Luckily, it's very easy to modify the the example vertex layout
+shown in the previous tutorial to work with our new texture coordinates.
+
+In our `VertexAttribPointer` calls, we declared an `aPosition` attribute with a size of 3 floats and a stride of 12 bytes (`3 * sizeof(float)` equals 12).
+Because of this, our buffer is being read like this:
+
+![Buffer reading wrong data](../../../images/opengl/chapter1/lesson3/wrong-pointers.png)
+
+As you can see, the vertex buffer is being read as if the first vertex was composed of the first three floats in the buffer, then the second
+vertex the next three floats, the third vertex the next three, and so on and so forth. However, this is now wrong! Since each
+vertex has two UV floats after its three position floats, this means that the first vertex's U and V floats are being read as the second
+vertex X and Y position floats!
+
+
+To fix this, we just need to go back to our `VertexAttribPointer()`, which is below our shader compiling section, and
+change it to:
+```c#
+//                       3 floats for position + 2 floats for texture coordinates! \/
+_gl.VertexAttribPointer(positionLoc, 3, VertexAttribPointerType.Float, false, 5 * sizeof(float), (void*)0);
+```
+
+This will make the buffer be read like this (including the texture coordinate pointer, which we'll add into our code later):
+
+![Buffer reading right data](../../../images/opengl/chapter1/lesson3/right-pointers.png)
+
+Now the positions of each vertex are being read correctly! Each position is still three floats, but by skipping the two floats after each position,
+we skip over the U and V floats in between each XYZ floats, thus reading them correctly.
+
+If we run the program now, we can see our quad again!
+
+## Using texture coordinates
+Now, we just need to prepare the shader to receive the texture coordinate values. For that, we need to modify both our shaders.
+
+Vertex Shader:
+```
+#version 330 core
+
+layout (location = 0) in vec3 aPosition;
+// Add a new input attribute for the texture coordinates
+layout (location = 1) in vec2 aTextureCoord;
+
+// Add an output variable to pass the texture coordinate to the fragment shader
+// This variable stores the data that we want to be received by the fragment
+out vec2 frag_texCoords;
+
+void main()
+{
+    gl_Position = vec4(aPosition, 1.0);
+    // Assigin the texture coordinates without any modification to be recived in the fragment
+    frag_texCoords = aTextureCoord;
+}";
+```
+in this case, we are putting manually the `location` of `aTextureCoord` as 1, because we already have a attribute in the location 0.
```

Comment:
```suggestion
In this case, we are manually specifying the `location` of `aTextureCoord` as 1, because we already have an attribute in the location 0.
```